packages/core/src/synthesis/Search.ts

```diff
@@ -0,0 +1,395 @@
+import {
```

Comment:
Since this code is WIP I didn't really take a close look at this file, lmk if you would like me to look at anything in particular!

Reply:
Nothing for now! Will continue this effort in another PR :D.